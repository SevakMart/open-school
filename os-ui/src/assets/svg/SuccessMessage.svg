<svg xmlns="http://www.w3.org/2000/svg" width="508" height="210" viewBox="0 0 508 210" fill="none">
  <g filter="url(#filter0_d_3187_2287)">
    <path d="M0 40C0 17.9086 17.9086 0 40 0H468C490.091 0 508 17.9086 508 40V48.7203C508 65.0552 498.068 79.7488 482.91 85.8376L273.108 170.114C263.694 173.896 253.194 173.959 243.734 170.29L25.5359 85.663C10.1433 79.693 0 64.8795 0 48.3697L0 40Z" fill="#EEEFF4"/>
  </g>
  <defs>
    <filter id="filter0_d_3187_2287" x="-35" y="-33" width="578" height="242.997" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB">
      <feFlood flood-opacity="0" result="BackgroundImageFix"/>
      <feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"/>
      <feOffset dy="2"/>
      <feGaussianBlur stdDeviation="17.5"/>
      <feComposite in2="hardAlpha" operator="out"/>
      <feColorMatrix type="matrix" values="0 0 0 0 0.85098 0 0 0 0 0.858824 0 0 0 0 0.913725 0 0 0 0.24 0"/>
      <feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow_3187_2287"/>
      <feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow_3187_2287" result="shape"/>
    </filter>
  </defs>
</svg>