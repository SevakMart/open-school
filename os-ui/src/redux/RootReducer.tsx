import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
