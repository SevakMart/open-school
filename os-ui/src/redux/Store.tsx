import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
