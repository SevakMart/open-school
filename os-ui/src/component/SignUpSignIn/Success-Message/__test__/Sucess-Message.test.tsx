import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../redux/Store';
import SuccessMessage from '../Success-Message';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test case for Successful message component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SuccessMessage
          message="You have successfuly registered"
          isSignUpSuccessfulRegistration
          isResendVerificationEmailMessage={false}
          isResetPasswordSuccessfulMessage={false}
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
