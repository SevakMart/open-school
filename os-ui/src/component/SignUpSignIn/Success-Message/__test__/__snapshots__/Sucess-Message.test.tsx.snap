// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test case for Successful message component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="closeIcon"
  >
    âœ–
  </div>
  <div
    class="mainContainer"
  >
    <h1
      class="title"
    >
      messages.successHeader
    </h1>
    <h2
      class="subtitle"
    >
      You have successfuly registered
    </h2>
  </div>
</DocumentFragment>
`;
